test/jest/types.spec.ts

```diff
@@ -0,0 +1,593 @@
+            );
+            expect(getName(ProductRefund)).toBe('Product Refunded');
+            expect(getName(PromotionView)).toBe('Promotion View');
+            expect(getName(PromotionClick)).toBe('Promotion Click');
+            expect(getName(ProductAddToWishlist)).toBe(
+                'Product Added to Wishlist'
+            );
+            expect(getName(ProductRemoveFromWishlist)).toBe(
+                'Product Removed from Wishlist'
+            );
+            expect(getName(ProductImpression)).toBe('Product Impression');
+        });
+        
+
+        it('returns other if the event type is not found', () => {
+            const { getName } = EventType;
+
+            expect(getName('foo' as unknown as number)).toBe('Other');
+        });
+
+        it('returns other if the commerce event type is not found', () => {
+            const { getName } = EventType;
+
+            expect(getName(NaN)).toBe('Other');
+        });
+    });
+});
+
+describe('CommerceEventType', () => {
+    const {
+        ProductAddToCart,
+        ProductRemoveFromCart,
+        ProductCheckout,
+        ProductCheckoutOption,
+        ProductClick,
+        ProductViewDetail,
+        ProductPurchase,
+        ProductRefund,
+        PromotionView,
+        PromotionClick,
+        ProductAddToWishlist,
+        ProductRemoveFromWishlist,
+        ProductImpression,
+    } = CommerceEventType;
+
+    it('returns a commerce event type', () => {
+        expect(ProductAddToCart).toEqual(10);
+        expect(ProductRemoveFromCart).toEqual(11);
+        expect(ProductCheckout).toEqual(12);
+        expect(ProductCheckoutOption).toEqual(13);
+        expect(ProductClick).toEqual(14);
+        expect(ProductViewDetail).toEqual(15);
+        expect(ProductPurchase).toEqual(16);
+        expect(ProductRefund).toEqual(17);
+        expect(PromotionView).toEqual(18);
+        expect(PromotionClick).toEqual(19);
+        expect(ProductAddToWishlist).toEqual(20);
+        expect(ProductRemoveFromWishlist).toEqual(21);
+        expect(ProductImpression).toEqual(22);
+    });
+
+    describe('#getName', () => {
+
+    });
+});
+
+describe('IdentityType', () => {
+    const {
+        Other,
+        CustomerId,
+        Facebook,
+        Twitter,
+        Google,
+        Microsoft,
+        Yahoo,
+        Email,
+        FacebookCustomAudienceId,
+        Other2,
+        Other3,
+        Other4,
+        Other5,
+        Other6,
+        Other7,
+        Other8,
+        Other9,
+        Other10,
+        MobileNumber,
+        PhoneNumber2,
+        PhoneNumber3,
+    } = IdentityType;
+
+    it('returns an identity type', () => {
+        expect(Other).toEqual(0);
+        expect(CustomerId).toEqual(1);
+        expect(Facebook).toEqual(2);
+        expect(Twitter).toEqual(3);
+        expect(Google).toEqual(4);
+        expect(Microsoft).toEqual(5);
+        expect(Yahoo).toEqual(6);
+        expect(Email).toEqual(7);
+
+        // There is no value for 8
```

Comment:
Context - 8 used to be `Alias` and it was a concept prior to Identity being implemented. After Identity was implemented, aliasing users was a new concept which supplanted the identity type.